### PR TITLE
grpc 1.55.0 protobuf 22.5

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
Needs a new Protobuf from #123791.

**Comment by carlocab** [comment]
Build failure while building static `libprotobuf`: https://github.com/Homebrew/homebrew-core/actions/runs/5026186180/jobs/9014275450

Removing it for now.

**Comment by carlocab** [comment]
Lots of failures. We may want to try Protobuf 22 first.

File: Formula/brpc.rb [hunk 1]
```diff
@@ -23,7 +24,9 @@ class Brpc < Formula
   depends_on "protobuf"
 
   def install
+    # Keep C++ standard in sync with abseil.rb
     args = %w[
+      -DCMAKE_CXX_STANDARD=17
```
Comment (carlocab) [inline Formula/brpc.rb:29]: Also needs to be set via `inreplace`.

File: Formula/libpulsar.rb [hunk 2]
```diff
@@ -27,8 +28,10 @@ class Libpulsar < Formula
   uses_from_macos "curl"
 
   def install
+    # Keep C++ standard in sync with abseil.rb
     system "cmake", ".", *std_cmake_args,
                     "-DBUILD_TESTS=OFF",
+                    "-DCMAKE_CXX_STANDARD=17",
```
Comment (carlocab) [inline Formula/libpulsar.rb:34]: Probably also needs to be forced somehow, since it's still building with C++11.

File: Formula/mavsdk.rb [hunk 3]
```diff
@@ -90,12 +91,13 @@ def install
...
     ) do
       system "tools/generate_from_protos.sh"
-
+      # Keep C++ standard in sync with abseil.rb
       # Source build adapted from
       # https://mavsdk.mavlink.io/develop/en/contributing/build.html
       system "cmake", *std_cmake_args,
                       "-Bbuild/default",
                       "-DSUPERBUILD=OFF",
+                      "-DCMAKE_CXX_STANDARD=17",
```
Comment (carlocab) [inline Formula/mavsdk.rb:100]: Already builds with C++17, so I guess it just needs an older `protobuf`.

File: Formula/protobuf-c.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ class ProtobufC < Formula
   url "https://github.com/protobuf-c/protobuf-c/releases/download/v1.4.1/protobuf-c-1.4.1.tar.gz"
   sha256 "4cc4facd508172f3e0a4d3a8736225d472418aee35b4ad053384b137b220339f"
   license "BSD-2-Clause"
-  revision 1
+  revision 2
```
Comment (carlocab) [inline Formula/protobuf-c.rb:7]: https://github.com/protobuf-c/protobuf-c/issues/544

File: Formula/osm-pbf.rb [hunk 5]
```diff
@@ -23,7 +23,8 @@ class OsmPbf < Formula
   uses_from_macos "zlib"
 
   def install
-    system "cmake", ".", *std_cmake_args
+    # Keep C++ standard in sync with abseil.rb
+    system "cmake", ".", "-DCMAKE_CXX_STANDARD=17", *std_cmake_args
```
Comment (carlocab) [inline Formula/osm-pbf.rb:27]: Still seems to be setting `-std=gnu++11` for some reason.

**Comment by carlocab** [comment]
Probably need `protobuf@21` and `grpc@1.54` 🫠

**Comment by carlocab** [comment]
`monero` and `wownero` aren't actually linking with protobuf. They either need an older protobuf or need to compile with C++17.

**Comment by carlocab** [comment]
Oops, forgot to fix `monero` and `wownero`. Applied some other fixes though. Will add some more fixes after this CI run to avoid breaking the cache.

File: Formula/mavsdk.rb (hunk 3 above)
Comment (carlocab) [reply Formula/mavsdk.rb:100]:
```suggestion
```
I should probably remove this to use whatever upstream is using.

File: Formula/protobuf-c.rb [hunk 6]
```diff
@@ -18,7 +18,7 @@ class ProtobufC < Formula
   end
 
   depends_on "pkg-config" => :build
-  depends_on "protobuf"
+  depends_on "protobuf@21"
```
Comment (carlocab) [inline Formula/protobuf-c.rb:21]:
```suggestion
  depends_on "protobuf@21" https://github.com/protobuf-c/protobuf-c/issues/544
```

File: Formula/monero.rb [hunk 7]
```diff
@@ -37,7 +38,11 @@ class Monero < Formula
   conflicts_with "wownero", because: "both install a wallet2_api.h header"
 
   def install
-    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    # Keep this in sync with C++ standard in abseil.rb
+    abseil_cxx_standard = 17
+    inreplace_files = %w[CMakeLists.txt cmake/CheckTrezor.cmake]
+    inreplace inreplace_files, /(CMAKE_CXX_STANDARD(?:\s|=))\d{2}/, "\\1#{abseil_cxx_standard}"
+    system "cmake", "-S", ".", "-B", "build", "-DCMAKE_CXX_STANDARD=#{abseil_cxx_standard}", *std_cmake_args
```
Comment (carlocab) [inline Formula/monero.rb:45]: Didn't work. I guess we need `protobuf@21` here.

File: Formula/opentelemetry-cpp.rb [hunk 8]
```diff
@@ -17,6 +18,7 @@ class OpentelemetryCpp < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "abseil" # already a dep of protobuf
```
Comment (carlocab) [inline Formula/opentelemetry-cpp.rb:21]: Needs `protobuf@21`.

**Comment by fxcoudert** [comment]
@carlocab for some reason even if only one Linux deps runner is still going on, the workflow is adding back the label I tried to remove

File: Formula/apache-arrow.rb [hunk 9]
```diff
@@ -42,7 +43,9 @@ def install
     # https://github.com/Homebrew/homebrew-core/issues/76537
     ENV.runtime_cpu_detection if Hardware::CPU.intel?
 
+    # Keep C++ standard in synch with abseil.rb
     args = %W[
+      -DCMAKE_CXX_STANDARD=17
```
Comment (carlocab) [inline Formula/apache-arrow.rb:48]: Should probably remove this for now.

**Review by carlocab — APPROVED** [review]
I don't think CI will get any better here. Will fix failures in a follow-up.

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by carlocab** [comment]
Thanks @ZhongRuoyu!

**Comment by carlocab** [comment]
Post-merge checklist (since GitHub keeps hiding it): https://github.com/Homebrew/homebrew-core/pull/131405#issuecomment-1578550900

**Pull request merged**